pyface/data_view/data_formats.py

```diff
@@ -0,0 +1,293 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+import csv
+from functools import partial
+from io import BytesIO, StringIO
+import os
+import json
+
+from .i_data_wrapper import DataFormat, text_format
+
+
+# Scalar formats
+
+def to_json(data, default=None):
+    """ Serialize an object to a JSON bytestring.
+
+    Parameters
+    ----------
+    data : any
+        The data to be serialized.
+    default : callable or None
+        Callable that takes a Python object and returns a JSON-serializable
+        data structure.
+
+    Returns
+    -------
+    raw_data : bytes
+        The serialized data as a bytestring.
+    """
+    str_data = json.dumps(data, default=default, separators=(',', ':'))
```

Comment:
Maybe.  Separators here is just to produce the compact form, rather than a human readable one.

Encoding is more difficult, as it may be possible to specify encodings as part of the mimetype but I'm not sure what the toolkits do with that, so for now I'm assuming UTF-8 for everything as a sensible default, and if we run into instances where it is a problem we can revisit.

Reply:
I think I'm leaving this as-is for now.  If this doesn't fit someone's use-case it's supposed to be easy to define your own data formats, and I expect that this module may grow additional formats and arguments to existing formats as time goes on.